src/OrbitQt/FilterPanelWidget.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "FilterPanelWidget.h"
+
+#include <QFontMetrics>
+#include <QFrame>
+#include <QWidget>
+
+#include "orbitmainwindow.h"
+#include "ui_FilterPanelWidget.h"
+
+FilterPanelWidget::FilterPanelWidget(QWidget* parent)
+    : QFrame(parent), ui(new Ui::FilterPanelWidget) {
+  ui->setupUi(this);
+
+  connect(ui->filterTracks, &QLineEdit::textChanged, this,
+          &FilterPanelWidget::FilterTracksTextChanged);
+  connect(ui->filterFunctions, &QLineEdit::textChanged, this,
+          &FilterPanelWidget::FilterFunctionsTextChanged);
```

Comment:
Nitpick, sorry, didn't spot this before: You could to this in the Designer right away, it was done like this before in the main window. You can define custom slots for your Widget in Designer, and then use the signal / slot editor to connect them.

But we're doing this in large parts in code already, no urgent need to change it, so ignore this if you want. Just wanted to let you know this is possible. In our case, I don't think we have a super strong need to do this because we're building parts of the UI dynamically and need to make the connections in Code. For a window completely defined in Designer, I find this to be easier because everything is done in one place.

Reply:
Thanks for the suggestion! I move these connections to the ui file. I totally agree that for a window defined in qtdesigner, it would be better if the connections are added in the same place.